Improve information for running code without internet connectivity (#2405)
viamrobotics/docs: docs/build/program/run.md

{{< readfile "/static/include/program/authenticate.md" >}}

> ## Run code remotely

Reviewer: I'm noticing that this section could also use improvement. I'm happy to make a ticket unless you want to roll it into here.

Author: ticket pls

This is useful because as long as that computer is able to establish a network connection with the machine's computer, your control logic will be executed on the machine.

## Run code on local network

Your machines do not need to be connected to the Internet for you to be able to run code.

> As long as your machine is connected to the same LAN or WAN network as the device running the code, you can connect to it and are run code. 

Reviewer: "you" and "it" are ambiguous here
```suggestion
You can run code from any computer connected to the same LAN or WAN network as your {{< glossary_tooltip term_id="machine" text="machine" >}}.
```
If you don't like ^ this suggestion, be sure to fix typo:
```suggestion
As long as your machine is connected to the same LAN or WAN network as the device running the code, you can connect to it and run code. 
```

> When connecting to a machine using the connection code from the [code sample tab](/build/program/#hello-world-the-code-sample-tab), a [client session](/build/program/apis/sessions/) automatically uses the [most efficient route](/build/program/connectivity/) to connect to your machine, which means the favored route for commands will be used.

Reviewer: ```suggestion
When you use the connection code from the [code sample tab](/build/program/#hello-world-the-code-sample-tab), that code establishes a [client session](/build/program/apis/sessions/) that automatically uses the [most efficient route](/build/program/connectivity/) to send commands to your machine.
```

> If this connection is over LAN or WAN and you lose internet connectivity, the connection will thus remain.

Reviewer: Can you add more explanation of how the user would make it be over LAN or WAN instead of internet? Is it just that that would inherently be the most efficient route?

> If this connection is over LAN or WAN and you lose internet connectivity, the connection will thus remain.

Author: ```suggestion
That means when your the device your code runs on is on the same network as your machine, even if internet is available, the connection will choose the most efficient route and connect over LAN or WAN.
If you subsequently lose internet connectivity, but stay connected to LAN or WAN, the connection will thus remain.
```

## Run code on-machine

> In case you run [PID control loops](https://en.wikipedia.org/wiki/PID_controller) or your machines have intermittent or no network connectivity, you can ensure this does not interfere with the code's execution, by running the control code on the same board that is running `viam-server`.

Reviewer: ```suggestion
If you run [PID control loops](https://en.wikipedia.org/wiki/PID_controller) or your machines have intermittent or no network connectivity, you can ensure this does not interfere with the code's execution by running the control code on the same board that is running `viam-server`.
```
I know these words aren't new but they're grammatically improvable.

This is useful because as long as that computer is able to establish a network connection with the machine's computer, your control logic will be executed on the machine.

## Run code on local network

Your machines do not need to be connected to the Internet for you to be able to run code.
As long as your machine is connected to the same LAN or WAN network as the device running the code, you can connect to it and run code. 

When you use the connection code from the [code sample tab](/build/program/#hello-world-the-code-sample-tab), that code establishes a [client session](/build/program/apis/sessions/) that automatically uses the [most efficient route](/build/program/connectivity/) to send commands to your machine.

> That means when your the device your code runs on is on the same network as your machine, even if internet is available, the connection will choose the most efficient route and connect over LAN or WAN.

Reviewer: typo
```suggestion
That means that when the device your code runs on is on the same network as your machine, even if internet is available, the connection will choose the most efficient route and connect over LAN or WAN.
```
